### ISSUE TITLE
AWS Scalar Types

### ISSUE THREAD

**Issue by EtienneProthon** [async-graphql/async-graphql#924]
Hello and thanks for this awesome lib !

I am deploying on AWS Lambda and generate graphql schema from my struct to give it to AWS AppSync. 
I am trying to have only one struct that match the database and expose the data to the users like this:
```
#[derive(Debug, sqlx::FromRow, Deserialize, Serialize, SimpleObject)]
#[graphql(complex)]
#[serde(rename_all = "camelCase")]
pub struct MyData {
    pub id: Uuid,
    #[graphql(skip)]
    pub created_at: DateTime<Utc>,
    #[graphql(skip)]
    pub updated_at: DateTime<Utc>,
    pub name: String,
    pub language: String,
}
```
I have several issues with the AWS Scalar Types because they only allow this list:
`AWSDate, AWSDateTime, AWSEmail, AWSIPAddress, AWSJSON, AWSPhone, AWSTime, AWSTimestamp, AWSURL`

So for the date i used `NewType`but to don't break sqlx i need to skip the field inside my main struct and redefine them after for graphql:
```
#[derive(NewType, Debug, Deserialize, Serialize)]
#[graphql(name = "AWSDateTime")]
pub struct AwsDateTime(NaiveDateTime);


#[ComplexObject]
impl MyData {
    async fn created_at(&self) -> AwsDateTime {
        self.created_at.naive_utc().into()
    }
    async fn updated_at(&self) -> AwsDateTime {
        self.updated_at.naive_utc().into()
    }
}
```
AWS doesn't also support UUID so i will need to do the same for the id field and expose it as a string.

Is there a better way to do this ? (i am wondering if i should create new struct just for graphql and implement From trait).

**Comment by sunli829** [comment]
> i am wondering if i should create new struct just for graphql and implement From trait

Usually I do that too, although there's a lot of boilerplate, I even wrote some macros to do this. 🙂

**Comment by frederikhors** [comment]
I think more complex examples are needed and can be of benefit both for those approaching for the first time and for those who want to improve.

@EtienneProthon do you want to help us with a complex example PR?

**Comment by EtienneProthon** [comment]
Yes i can help you, i use async-graphql for some time right now with different architectures (Serverless, Docker).
Also i just found that if i just change the type on the code that generate the graphql schema i don't need any mapping on my side because on serverless i only use the lib to generate the schema from my structs (i didn't find an easy way to use the handlers from lambda).

**Comment by EtienneProthon** [comment]
I am trying to create an exemple of integration with AWS Lambda and i saw that before there was a lambda integration inside this repository -> https://github.com/async-graphql/async-graphql/commit/191758f9b0b1765d9aa9d1bf69cf93d6793e804e
Is there any reason to have removed it ? Because this kind of integration would be very usefull.
I also find this repository that add integration lambda-http -> https://github.com/adriandelgado/async-graphql-lambda

**Comment by EtienneProthon** [comment]
Hey any more informations ? @sunli829 @frederikhors 
It's getting really complicated as i am trying to use cursor for this lib and i don't have an easy solution to call schema.execute with the input i have from the lambda.
If we can fix this issue i can create a complete exemple on how to integrate this lib with aws lambda (and add more exemples about cursor also).

**Comment by frederikhors** [comment]
Nope
